hermes/bin/src/runtime_extensions/hermes/http_gateway/gateway_task.rs

```diff
@@ -0,0 +1,140 @@
+//! HTTP Gateway task
+
+use std::{
+    collections::HashMap,
+    convert::Infallible,
+    net::SocketAddr,
+    sync::{Arc, Mutex},
+};
+
+use hyper::{
+    self,
+    server::{conn::AddrStream, Server},
+    service::{make_service_fn, service_fn},
+};
+use tracing::{error, info};
+
+use super::routing::router;
+
+/// HTTP Gateway port
+const GATEWAY_PORT: u16 = 5000;
+
+#[derive(Debug, Clone)]
+/// hostname (node name)
+pub(crate) struct Hostname(pub String);
```

Comment:
Is it necessary to have a `pub` visibility for this field ?
Is it possible to hide it ?

Reply:
They are currently shared between multiple files. We can encapsulate later as the connection manager evolves. It's just a placeholder for now.